api/database/boltdb/json.go

```diff
@@ -1,25 +1,126 @@
 package boltdb
 
 import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
 	"encoding/json"
+	"fmt"
+	"io"
 
 	jsoniter "github.com/json-iterator/go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 )
 
+var encryptedStringTooShort = fmt.Errorf("encrypted string too short")
+
 // MarshalObject encodes an object to binary format
-func MarshalObject(object interface{}) ([]byte, error) {
-	return json.Marshal(object)
+func MarshalObject(object interface{}, passphrase string) ([]byte, error) {
```

Comment:
I really wonder if we should just make these methods on `(connection *DbConnection)`, and thus don't need to pass the extra parameter.